docs/user/interactive-running-locally.rst

```diff
@@ -0,0 +1,121 @@
+.. _local_interactive_environments:
+
+Running RenkuLab Interactive Sessions on Your Own Machine
+=========================================================
+
+Each Renku project contains configuration files (``Dockerfile``,
+``requirements.txt`` etc.) for building reproducible session images. We build
+these automatically when you push your project to the server and store them in
+the image registry connected to the Renku instance that you are using. The
+custom-built images are used whenever you launch an interactive session on
+RenkuLab.
+
+You can of course also use these same images to run your code in the correct
+environment anywhere that `Docker <https://www.docker.com>`_ is available.
+
+.. note::
+
+    All of the instructions below assume that you have cloned your project
+    locally and that you have `Docker
+    installed <https://docs.docker.com/get-docker/>`_. On Linux systems, Docker
+    requires root privileges - if you do not have root access to the host, you
+    might consider `rootless
+    Docker <https://docs.docker.com/engine/security/rootless/>`_.
+
+
+Building your own image
+-----------------------
+
+If you are working on a project that has not yet been pushed to the server,
+or you just prefer to build the image locally, you can certainly do so
+with Docker. To build the image, run this command from the root directory
+of your project:
+
+.. code-block::
+
+    $ docker build -t <imageName>:<tag> .
+
+The ``imageName`` and ``tag`` can be anything you want, but you might want to
+try to use something you will remember. To see which images you have in
+your local Docker registry, run
+
+.. code-block::
+
+    $ docker images
+
+
+Using the images built on RenkuLab
+----------------------------------
+
+If your repository on RenkuLab is public, the easiest way to spin up your
+runtime environment locally is by pulling the image that has already been built.
+First, make sure that your project has been pushed and that the image build is
+complete. If you are not sure, the easiest way to check is to navigate to your
+project in RenkuLab and go to launch a new session - if the "Start environment"
+button is available, it means the image has been built already.
+
+To pull the image, you need to construct the image name. This is typically the
```

Comment:
There is an easier way to get that information from the UI, not sure if we want to include it since it requires launching the environment.
Anyway, once that is up and running, the user can click the green icon on the left to see the image URL

![Screenshot_20210325_094805](https://user-images.githubusercontent.com/43481553/112450859-1f316700-8d55-11eb-8047-508b0c4984d8.png)

Reply:
ah I forgot about that! Do you think we could add a button somewhere (maybe on the page listing the project commits) that would let you copy the image name? Also here it would be nice to be able to click to have it sent to the clipboard

Reply:
Adding the clipboard copy button is straightforward. Adding the same information in the commit list (you mean the one in the following screenshot?) is a bit more tedious since we need to verify that the entry in the registry exists, so I wouldn't do that for all the commits. We can add another button to request the URL for a specific commit, but it won't be very intuitive.
I'm wondering if it makes sense to add than info (for the latest commit) in the `Settings` page :thinking: 
Once we support changing branch, that would cover most of the cases.

![Screenshot_20210325_135744](https://user-images.githubusercontent.com/43481553/112477046-da1c2d80-8d72-11eb-9a3d-03c060df9510.png)

Reply:
my initial thought was that if we just had a button to copy the image name, we would make clear that it doesn't on its own guarantee that the image is there.. but I can see that being annoying. It would be nice to not have to go to the settings page for this, but just have it readily available (along with a button to copy the renku clone command) from some front page of the project. Maybe the two locations in red in this screenshot?

![Screenshot 2021-03-25 at 14 11 31](https://user-images.githubusercontent.com/3353586/112478328-2f0c7380-8d74-11eb-9cbe-e82dc30806b0.png)

In any case, that's a bit off topic here, but if we're going to add something like this I would not complicate the instructions more at this point - it also requires the user to launch an environment first before they can see the image name, which is sort of annoying imho. 

btw, I would also be fine with your proposal to have that on the settings page, although it doesn't really sound like a setting to me.

Reply:
> In any case, that's a bit off topic here

Sure, I can open an issue in the UI repo to track this

> it doesn't really sound like a setting to me

That's true, I guess the setting page is not the place.
I wouldn't put this info next to the project name, it seems too prominent, and it would require polling the whole time.
I think an overview tab would work well.
Actually, the `Status` may be a good place. I'm also fine with creating a new one `Registry`. Anyway, let's move this discussion to the UI, I'll link here the new issue.

Reply:
SwissDataScienceCenter/renku-ui#1289

Reply:
haha you beat me to it! thanks!